core/src/smartcontracts/isi/asset.rs

```diff
@@ -1,7 +1,10 @@
 //! This module contains [`Asset`] structure, it's implementation and related traits and
 //! instructions implementations.
 
-use iroha_data_model::prelude::*;
+use iroha_data_model::{
```

Comment:
Minor comments about this file's layout. You should probably move the macro invocations to the top of `mod isi`, this is what people will mainly be interested in. Then the `macro_rules` for mint, burn and transfer, with the assertions at the bottom. 

Alternatively, to keep things tidy, you could also put the implementations into `mod mint`, `mod burn` and `mod transfer`, so the business logic related to each operation is in one place. The previous comment holds, (impl invocations first, then macro, then asserts at bottom). 

This is not a hard requirement for approval, just a recommendation for how to organise code.

Reply:
I agree it will be better